draft-ietf-acme-acme.md

```diff
@@ -376,18 +376,47 @@ multiple JWS serialization formats. When sending an ACME request
 with a non-empty body, an ACME client implementation MUST use the 
 HTTP Content-Type {{!RFC7231}} header to indicate which JWS serialization format 
 is used for encapsulating the ACME request payload.
-The following Content-Type values may be used for this purpose:
+
+The list below describes the currently defined Content-Type header 
+values for indicating which JWS serialization format is used,
+along with the associated requirements. 
+A "root JWS" is the JWS used to encapsulate 
+the entire ACME request payload, and a "nested JWS" 
+is a JWS contained within the ACME request payload
+(such as the "externalAccountBinding" described in 
+{{external-account-binding}} 
+or the "key-change" object described in {{account-key-roll-over}}). 
+One of the following Content-Type header values MUST be used to indicate 
+the JWS serialization format:
 
  - "application/jose":
-   - The JWS Compact Serialization {{!RFC7515}} MUST be used. 
-   - The JWS Payload MUST NOT be detached. 
-   - The JWS Unencoded Payload Option {{!RFC7797}} MUST NOT be used. 
+   - Each root JWS MUST meet the following criteria:
+     - The JWS Compact Serialization {{!RFC7515}} MUST be used. 
+     - The JWS Payload MUST NOT be detached. 
+     - The JWS Unencoded Payload Option {{!RFC7797}} MUST NOT be used.
+   - Each nested JWS MUST be a JSON string that meets the following criteria:
+     - The JWS Compact Serialization {{!RFC7515}} MUST be used. 
+     - The JWS Payload MUST NOT be detached. 
+     - The JWS Unencoded Payload Option {{!RFC7797}} MUST NOT be used.
  - "application/jose+json":
-   - Either the JWS Flattened JSON Serialization {{!RFC7515}} 
-     or the JWS General JSON Serialization {{!RFC7515}} MUST be used. 
-   - The JWS Payload MUST NOT be detached. 
-   - The JWS Unencoded Payload Option {{!RFC7797}} MUST NOT be used.
-   - The JWS Unprotected Header {{!RFC7515}} SHOULD NOT be used. 
+   - Each root JWS MUST meet the following criteria:
+     - Either the JWS Flattened JSON Serialization {{!RFC7515}} 
+       or the JWS General JSON Serialization {{!RFC7515}} MUST be used. 
+     - The JWS Payload MUST NOT be detached. 
+     - The JWS Unencoded Payload Option {{!RFC7797}} MUST NOT be used.
+     - The JWS Unprotected Header {{!RFC7515}} SHOULD NOT be used. 
+   - Each nested JWS MUST be a JSON object that meets the following criteria:
+     - Either the JWS Flattened JSON Serialization {{!RFC7515}} 
+       or the JWS General JSON Serialization {{!RFC7515}} MUST be used. 
+     - The JWS Payload MUST NOT be detached. 
+     - The JWS Unencoded Payload Option {{!RFC7797}} MUST NOT be used.
+     - The JWS Unprotected Header {{!RFC7515}} SHOULD NOT be used. 
+
+If a server receives a request using a serialization it does not
```

Comment:
see list or #398

Reply:
Should I change this paragraph to read:

If a server receives a request using a serialization it does not
support, then it MUST send a response with HTTP status code 415
(Unacceptable Media Type) and a problem document 
with error type "unsupportedSerialization". This problem document MUST contain a "supportedSerializations" array of strings indicating the acceptable serialization content types.

Should there be an addition to the directory object (or somewhere within it) to help prevent this issue?

Reply:
I think better to continue this on the list; https://www.ietf.org/mail-archive/web/acme/current/msg02618.html and follow-ups.